connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -120,7 +120,7 @@ import org.apache.spark.util.Utils
  */
 class Dataset[T] private[sql] (
     val sparkSession: SparkSession,
-    private[sql] val plan: proto.Plan,
+    val plan: proto.Plan,
```

Comment:
Can we mark this field as DeveloperApi. Not sure if we can do that at the field level though.

Reply:
It's possible. It's also done for the non-Connect `Dataset`: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L191